### ISSUE TITLE
Error while compiling openmpi with opkgc

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#401]
**Reported by dikim on 21 Aug 2007 20:00 UTC**
Hi,

I have tried to compile openmpi with opkgc on RHEL-5 (x86) and I thing that there is an error here.
```
[openmpi](root@oscar-rhel5)# opkgc --dist=rhel -v -v
[Read config file /etc/opkgc.conf
[DEBUG](INFO]) /usr/share/opkgc/opkg.xsd loaded
[./config.xml loaded
[INFO](DEBUG]) Validating XML file against schema
[Package name: openmpi
[DEBUG](DEBUG]) Remove file: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/packages/openmpi/config.xml
[Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/packages/openmpi
[DEBUG](DEBUG]) Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/packages
[Remove file: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing/cpi.c
[DEBUG](DEBUG]) Remove file: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing/f77hello.f
[Remove file: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing/rm_script.openmpi
[DEBUG](DEBUG]) Remove file: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing/test_user
[Remove file: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing/cxxhello.cc
[DEBUG](DEBUG]) Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing
[Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi
[DEBUG](DEBUG]) Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing
[Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar
[DEBUG](DEBUG]) Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib
[Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/var
[DEBUG](DEBUG]) Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/usr/share/doc/opkg-openmpi/doc
[Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/usr/share/doc/opkg-openmpi
[DEBUG](DEBUG]) Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/usr/share/doc
[Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/usr/share
[DEBUG](DEBUG]) Remove dir: /usr/src/redhat/BUILD/opkg-openmpi/usr
[Remove dir: /usr/src/redhat/BUILD/opkg-openmpi
[INFO](DEBUG]) Create dir: /usr/src/redhat/BUILD/opkg-openmpi/usr/share/doc/opkg-openmpi/doc
[Create dir: /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing
[INFO](INFO]) Copy ./testing/rm_script.openmpi to /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing
[Copy ./testing/cxxhello.cc to /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing
[INFO](INFO]) Copy ./testing/f77hello.f to /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing
[Copy ./testing/cpi.c to /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing
[INFO](INFO]) Copy ./testing/test_user to /usr/src/redhat/BUILD/opkg-openmpi/var/lib/oscar/testing/openmpi/testing
[Generates /usr/src/redhat/SPECS/opkg-openmpi.spec from template /usr/share/opkgc/tmpl/rpm/opkg.spec.tmpl
[DEBUG](INFO]) Execute: rpmbuild -ba /usr/src/redhat/SPECS/opkg-openmpi.spec
[Processing files: opkg-openmpi-1:1.2.3-1
[DEBUG](DEBUG]) Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
[Requires: /bin/sh
[DEBUG](DEBUG]) Processing files: opkg-openmpi-server-1:1.2.3-1
[Processing files: opkg-openmpi-client-1:1.2.3-1
[DEBUG](DEBUG]) Checking for unpackaged file(s): /usr/lib/rpm/check-files /usr/src/redhat/BUILD/opkg-openmpi
[Wrote: /usr/src/redhat/SRPMS/opkg-openmpi-1:1.2.3-1.src.rpm
[DEBUG](DEBUG]) Wrote: /usr/src/redhat/RPMS/i386/opkg-openmpi-1:1.2.3-1.i386.rpm
[Wrote: /usr/src/redhat/RPMS/i386/opkg-openmpi-server-1:1.2.3-1.i386.rpm
[DEBUG](DEBUG]) Wrote: /usr/src/redhat/RPMS/i386/opkg-openmpi-client-1:1.2.3-1.i386.rpm
[Packages succesfully generated
Traceback (most recent call last):
  File "/usr/bin/opkgc", line 149, in ?
    main()
  File "/usr/bin/opkgc", line 146, in main
    c.compile(build)
  File "/usr/lib/python2.4/site-packages/Opkgc/OpkgcCompiler.py", line 246, in compile
    self.getDestDir())
  File "/usr/lib/python2.4/shutil.py", line 81, in copy
    copyfile(src, dst)
  File "/usr/lib/python2.4/shutil.py", line 47, in copyfile
    fsrc = open(src, 'rb')
IOError: [Errno 2](INFO]) No such file or directory: '/usr/src/redhat/RPMS/noarch/opkg-openmpi-1:1.2.3-1.noarch.rpm'
```

As you can see, the meta rpms are generated on RPMS/i386 but opkgc tries to look for it at RPMS/noarch.[[BR]]
I think that opkgc could not use the right path to find the right meta rpms.

Regards,

- DongInn

**Comment by dikim33** [comment]
**Comment by jparpail on 24 Aug 2007 06:59 UTC**
Trying to investigate and solve this ASAP.

The error comes from the package is not built for the right architecture. Waiting for the fix, you can pick manually the packages from /usr/src/redhat/RPMS/i386/ dir.

Jean

**Comment by dikim33** [comment]
**Comment by jparpail on 6 Sep 2007 11:50 UTC**
(In [[OPKGC](6150]))
	Let generated RPM packages in the RPMS dir (given by macro 
	%_rpmdir) and print this information.
	Closes #401
